src/Database/AggregatesBuilder.php

```diff
@@ -0,0 +1,149 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         4.1.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database;
+
+use Cake\Database\Expression\AggregateExpression;
+
+/**
+ * Helper for generating AggregateExpressions which are
+ * SQL functions that can be used as aggregate or window functions.
+ */
+class AggregatesBuilder
+{
+    /**
+     * Returns a AggregateExpression representing a call to SQL SUM function.
+     *
+     * @param mixed $expression the function argument
+     * @param array $types list of types to bind to the arguments
+     * @return \Cake\Database\Expression\AggregateExpression
+     */
+    public function sum($expression, $types = []): AggregateExpression
```

Comment:
Would the planned window related methods be available on the sum() expression?

Reply:
Yes, any supported aggregate function can be extend to a window function.

Even without that, we can add FILTER support to this interface for non-window functions.